pkg/interfaces/contracts/pool-utils/IManagedPool.sol

```diff
@@ -0,0 +1,316 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "../solidity-utils/openzeppelin/IERC20.sol";
+import "../vault/IBasePool.sol";
+
+interface IManagedPool is IBasePool {
+    event GradualSwapFeeUpdateScheduled(
+        uint256 startTime,
+        uint256 endTime,
+        uint256 startSwapFeePercentage,
+        uint256 endSwapFeePercentage
+    );
+    event GradualWeightUpdateScheduled(
+        uint256 startTime,
+        uint256 endTime,
+        uint256[] startWeights,
+        uint256[] endWeights
+    );
+    event SwapEnabledSet(bool swapEnabled);
+    event MustAllowlistLPsSet(bool mustAllowlistLPs);
+    event AllowlistAddressAdded(address indexed member);
+    event AllowlistAddressRemoved(address indexed member);
+    event ManagementAumFeePercentageChanged(uint256 managementAumFeePercentage);
+    event ManagementAumFeeCollected(uint256 bptAmount);
+    event CircuitBreakerSet(
+        IERC20 indexed token,
+        uint256 bptPrice,
+        uint256 lowerBoundPercentage,
+        uint256 upperBoundPercentage
+    );
+    event TokenAdded(IERC20 indexed token, uint256 normalizedWeight);
+    event TokenRemoved(IERC20 indexed token);
+
+    /**
+     * @notice Returns the effective BPT supply.
+     *
+     * @dev The Pool owes debt to the Protocol and the Pool's owner in the form of unminted BPT, which will be minted
+     * immediately before the next join or exit. We need to take these into account since, even if they don't yet exist,
+     *  they will effectively be included in any Pool operation that involves BPT.
+     *
+     * In the vast majority of cases, this function should be used instead of `totalSupply()`.
+     */
+    function getActualSupply() external view returns (uint256);
+
+    // Swap fee percentage
+
+    /**
+     * @notice Schedule a gradual swap fee update.
+     * @dev The swap fee will change from the given starting value (which may or may not be the current
+     * value) to the given ending fee percentage, over startTime to endTime.
+     *
+     * Note that calling this with a starting swap fee different from the current value will immediately change the
+     * current swap fee to `startSwapFeePercentage`, before commencing the gradual change at `startTime`.
```

Comment:
It does change immediately, because the interpolation will now return the start value. It used to set it in BasePool and emit SwapFeePercentageChanged, but since all that no longer exists, no need to talk about past functionality.

Reply:
I'm not sure I follow, isn't that what the docs say?

Reply:
Yes; I was just arguing with myself about whether it should explain how it changes immediately. I think it's fine as is.